core/runtime/src/main/java/io/quarkus/runtime/logging/SocketConfig.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.runtime.logging;
+
+import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.logging.Level;
+
+import org.jboss.logmanager.handlers.SocketHandler.Protocol;
+
+import io.quarkus.runtime.annotations.ConfigDocMapKey;
+import io.quarkus.runtime.annotations.ConfigDocSection;
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class SocketConfig {
+
+    /**
+     * If socket logging should be enabled
+     */
+    @ConfigItem
+    boolean enable;
+
+    /**
+     *
+     * The IP address and port of the server receiving the logs
+     */
+    @ConfigItem(defaultValue = "localhost:4560")
+    InetSocketAddress endpoint;
+
+    /**
+     * Sets the protocol used to connect to the server
+     */
+    @ConfigItem(defaultValue = "tcp")
+    Protocol protocol;
+
+    /**
+     * Enables or disables blocking when attempting to reconnect a
+     * {@link Protocol#TCP
+     * TCP} or {@link Protocol#SSL_TCP SSL TCP} protocol
+     */
+    @ConfigItem
+    boolean blockOnReconnect;
+
+    /**
+     * The log message formatter. json or pattern is supported
+     */
+    @ConfigItem(defaultValue = "pattern")
+    String formatter;
```

Comment:
Why is this added vs what SysLogConfig?

Reply:
This is added as the output where you are sending messages can require formatted output differently than the Pattern. In our case we have logstash which has input tcp and json_lines. We need custom attributes/fields to add to the json file. If you have some other suggestion how to achieve it - I am happy to modify it.

Reply:
@gsmet WDYT about this? Does this use case warrant doing something different than what our SysLog handling does?

Reply:
Initially I thought I will use logging-json extension however this extension is limited to console logging. Maybe it would be more beneficial to extend that support instead and keep socket logging clean

Reply:
Yeah, I think that's a better approach

Reply:
HI @geoand, @gsmet ,
I have reworked the code completely. Now socket logger in core has just pattern and logging-json was extended to support all handlers not just console as it was. Now file, console, syslog, socket and also named handlers can log as json. Previous implementation of consoleFormatter was opiniated and if JSON was added as extension basic console but also all named handlers were forced to log as JSON. I kept the original behavior but personally I am not fan of it. With "json=false"  you can set console to use PatternFormatter and some named console handlers to use JSON. But you cannot use console as JSON and some named console handler to use Pattern. Its just because implementation should be backward compatible.
If users add logging-json extension 
- all handlers will start log to JSON (even file,syslog,socket) . Not sure if its ok?
- Additional fields can be defined as quarkus.log.socket.json.additional-field.myownfield.value=testingvalue 
- Keyoverrides was exposed too

I changed integration test of logging-json as previous one was not doing anything.

Reply:
Thanks a lot @JozefDropco.

I'll have a look soon

Reply:
I somehow missed this PR but I wonder if the work I did in https://github.com/quarkusio/quarkus/pull/25505 isn't somehow related.
I probably need to have another look at it.